README.md

```diff
@@ -93,7 +93,13 @@ If the installation fails at the point where it installs `canvas`, you may want
 
 
 # Installation (Docker) 🐳
-[TO-DO]
+
+Assuming you have Docker already installed, the following steps should result in the application running locally on port 3000:
+
+1. Copy the appropriate `.env` file: `cp .env.development .env`
+2. Build the container: `docker-compose -f docker-compose-development.yml build`
+3. Install dependencies: `docker-compose -f docker-compose-development.yml run app yarn install --frozen-lockfile --production=false`
```

Comment:
This `yarn install` command is already there in the Dockerfile, so I don't think we need to run it separately.

Reply:
Hmm, I ran into troubles with relying on the one in the Dockerfile; can you try the following and let me know how it works for you?

```
sudo rm -rf node_modules/
docker-compose -f docker-compose-develop.yml down
docker-compose -f docker-compose-develop.yml build
docker-compose -f docker-compose-develop.yml up
```

When I do that, I get:
```
resource-watch | yarn run v1.22.5
resource-watch | $ cross-env NODE_ENV=development NODE_OPTIONS=--max_old_space_size=4096 node index.js
resource-watch | /bin/sh: cross-env: not found
resource-watch | error Command failed with exit code 127.
```

Running the `yarn install ...` command sets things right for me though.

Reply:
Ok, I think the reason this worked for me and not for you initially is that I spun up the app on host before doing it in docker, so I had a complete and recent `node_modules` outside the container.  Which is important because I think that's the one that's actually getting used by the container.  The `Dockerfile` copies lots of files in and runs `yarn install`, but then the `volumes:` mapping in `docker-compose-develop.yml` comes along and mounts the working directory from the host over top of that, so the one created when the container was built is masked over and if you didn't run `yarn install` on host, you suddenly don't have a `node_modules` directory.  And running `docker-compose run yarn install` solves the problem because it actually installs everything in the (mapped) host directory.

This seems like not an ideal setup, since it makes the container less self-contained in general and specifically for the current purposes where we want to support running the app with or without Docker, could create some confusing errors.
Here's a blog post that talks about how to deal with it: https://www.docker.com/blog/keep-nodejs-rockin-in-docker/
The second option (move `node_modules` up a directory inside the container) seems like the one we need, since the first is basically "make sure you never do anything outside the container".

There's also [this SO answer](https://stackoverflow.com/a/38601156), which suggests mapping `node_modules` to a persistent `docker-compose` volume, but that seems worse than the "move it up a level" option to me, since it introduces another entity that could become stale or wrong in potentially surprising ways.

Reply:
Okay, after a bit of fiddling, these approaches all seem to have some different benefits / drawbacks:
- The "empty bind mount" approach included as part of the second strategy gets pretty far along this path on its own. With the volumes configured this way, the container (mostly) doesn't pay attention to the state of `node_modules` on the host:
```yml
    volumes:
      - .:/usr/src/app
      # Prevent host node_modules from overwriting node_modules in the container
      - /usr/src/app/node_modules
```
(This is slightly different than what's suggested in the blog post, but that approach creates a volume, which causes its own share of headaches because it introduces persistence, which isn't necessarily what you want in your `node_modules` folder -- that threw me for a loop a couple times. This approach is described in the SO question, and it seems to work identically except that no volume is created, which seems simpler).
The one exception I've found to not paying attention to `node_modules` on the host is that it'll get upset if you completely delete `node_modules` on the host while the container is running. Also, if you run `docker-compose up` when `node_modules` doesn't already exist on the host, it'll create an empty, root-owned `node_modules` folder on the host, which will cause subsequent invocations of `yarn install` from the host to fail unless the root-owned `node_modules` folder is deleted first. Nonetheless, this approach does make `node_modules` inside the container mostly independent from `node_modules` on the host, so you can do container- and host-based development simultaneously, as long as you start with host-based first and don't mind restarting the container if you accidentally delete `node_modules` while it's running.
- Implementing the "move the node_modules folder up a level" approach solves the issue with deleting `node_modules`, but it adds some extra complexity to the Dockerfile, relies on a subtle detail of Node's module discovery algorithm, and still seems to suffer from the "empty root-owned `node_modules` on the host" problem.
- The status quo is essentially approach 1 described in the blog post, i.e. "do everything in the container". The container still contains the correct `node_modules` folder to function in a production or CI environment, but in development, the container's `node_modules` get overwritten by the host `node_modules`. So you have to choose between host- or container-based development, but that's not enforced anywhere. If you mess up and accidentally run `yarn install` from your host even though you're trying to do container-based development, things will probably mostly still work, but sometimes they might not and will probably fail in strange ways when they do fail.

Between these, I think I'm leaning toward the status quo--if there was a way to keep the empty `node_modules` from showing up then I think the "move `node_modules` up a level" solution would be more attractive, but I haven't found one so far, and otherwise the solution is more complicated and relies on behavior that's not entirely mainstream, without really changing the number of extant land mines (though perhaps changing their character enough that it matters).

Anyway, I've pushed commits with both approaches:
- dd614f1  implements just the empty bind mount
- 03880ef moves the `node_modules` folder up a level as well.

Take a look and let me know what you think!

Reply:
This is certainly more troublesome than I would have predicted.  Making an empty directory for the "empty bind mount" seems odd and wrong.  I tried a little more fiddling with it, e.g. "What happens if I mount the host's `node_modules` to somewhere else?" and "What if we bind to `/dev/null` instead of not providing a local path?"  The former is no different from just the empty bind mount, I think. The latter still creates `node_modules` but as an empty regular file, not a directory, which came as a surprise.  Still owned by root, so it's just as inconvenient.

I'm not necessarily opposed to "do everything in the container", but that's not how I would describe the status quo.  The README starts with instructions for running on host, then these Docker instructions are a ways down the page.  So that definitely implies that both are valid, and doesn't make it clear that they conflict and switching between them would cause problems.  For my sample size of one, I can say that the first thing I did was spin it up on host, spin up the container, then get confused about unexpected behavior caused by the interaction of the two.

So unless we're prepared to purge the local instructions and make docker-compose the only blessed dev environment, I think the "things will probably mostly still work, but sometimes they might not and will probably fail in strange ways when they do fail" is something we need to avoid.  The "empty `node_modules` directory owned by root" is annoying, but it seems clearly preferable to me, since it's hard to miss, the obvious fix is the right one (change the owner), and we can put a note about it in the README.

So yeah, I vote for the version with both of the fixup commits, and maybe also a note in this section of the README to say that if you try to spin up on host after first running it in a container, you might need to `chown` the `node_modules` directory.

Reply:
>  For my sample size of one, I can say that the first thing I did was spin it up on host, spin up the container, then get confused about unexpected behavior caused by the interaction of the two.

Hah, I definitely skipped past all the on-host instructions in the hope of finding something Docker related, and then just dove into the source when I couldn't find anything. But now that you bring it up, I agree that "willfully disregard all README instructions" is not the approach we should assume most people will take. :slightly_smiling_face: 

I've gone ahead and squashed the two fixups, and also added some additional README text to make the `node_modules` pitfall clearer and direct people a bit more forcefully to the Docker instructions. This is ready for another look!